src/python_testing/matter_testing_support.py

```diff
@@ -1492,19 +1492,35 @@ def parse_matter_test_args(argv: Optional[List[str]] = None) -> MatterTestConfig
 
     return convert_args_to_matter_config(parser.parse_known_args(argv)[0])
 
+def async_to_sync(awaitable):
+    "Call sync, the async function."
+    # Usage of new implementation of ChipDeviceController cause the necessary to not use asyncio.run,
+    # because the initialization of `asyncio.Lock` use the `asyncio.get_event_loop()` on init.
+    # Usage of `asyncio.run()` and `asyncio.get_event_loop()` are exclusive.
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(awaitable)
+
+def generate_random_nodeid(excluded_nodeid: typing.Optional[typing.List] = None) -> int:
```

Comment:
List is not ideal, maybe Set would be better.

Why not set the default to an empty set, and then you do not need to handle None and everything can go through the same `while True` branch.